### PR TITLE
Basic framework of local storage for node group

### PR DESCRIPTION
**Pull request by ray6080**

### CODE REVIEW

File: src/include/storage/local_storage.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+// Data structures in LocalStorage are not thread-safe.
+// For now, we only support single thread insertions and updates. Once we optimize them with
+// multiple threads, LocalStorage and its related data structures should be reworked to be
+// thread-safe.
+class LocalStorage {
+public:
+    LocalStorage(storage::StorageManager* storageManager, storage::MemoryManager* mm);
+
+    void scan(common::table_id_t tableID, common::ValueVector* nodeIDVector,
+        const std::vector<common::column_id_t>& columnIds,
```
Comment (acquamarin) [inline src/include/storage/local_storage.h:18]: columnIds->columnIDs?

File: src/include/storage/local_table.h [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    void prepareCommit();
+
+private:
+    std::map<common::property_id_t, std::unique_ptr<LocalColumn>> columns;
+    storage::NodeTable* table;
+};
+
+} // namespace storage
+} // namespace kuzu
```
Comment (acquamarin) [inline src/include/storage/local_table.h:121]: newline

File: src/storage/storage_manager.cpp [hunk 3]
```diff
@@ -46,6 +46,7 @@ std::unique_ptr<MetadataDAHInfo> StorageManager::createMetadataDAHInfo(
         metadataDAHInfo->childrenInfos.push_back(
             createMetadataDAHInfo(*VarListType::getChildType(&dataType)));
     } break;
+        // TODO(Guodong): We don't need NULL for string overflow, right?
```
Comment (acquamarin) [inline src/storage/storage_manager.cpp:49]: I guess so.

File: test/transaction/transaction_test.cpp [hunk 4]
```diff
@@ -282,3 +273,6 @@ TEST_F(TransactionTests, ExecuteWriteQueryInReadOnlyTrx) {
     ASSERT_EQ(
         result->getErrorMessage(), "Can't execute a write query inside a read-only transaction.");
 }
+
+} // namespace transaction
+} // namespace kuzu
```
Comment (acquamarin) [inline test/transaction/transaction_test.cpp:278]: newline

File: test/transaction/transaction_test.cpp [hunk 5]
```diff
@@ -245,7 +236,7 @@ TEST_F(TransactionTests, OpenReadOnlyTransactionTriggersTimeoutErrorForWriteTran
     // Note that TransactionTests starts 1 read and 1 write transaction by default.
     getTransactionManager(*database)->setCheckPointWaitTimeoutForTransactionsToLeaveInMicros(
         10000 /* 10ms */);
-    updateAgeAndEyeSightPropertiesForNodes0And1();
+    //    updateAgeAndEyeSightPropertiesForNodes0And1();
```
Comment (acquamarin) [inline test/transaction/transaction_test.cpp:239]: Why do we comment this update?

**Review by acquamarin — APPROVED** [review]

File: test/transaction/transaction_test.cpp (hunk 5 above)
Comment (ray6080) [reply test/transaction/transaction_test.cpp]: Reverted.

**Pull request merged**